woocommerce/woocommerce-ios
# [Orders with Coupons M5]Decouple Coupons extra logic from the list so it can be easily reused.

Part of: #10246 


## Description

With this PR we prepare to show the list of coupons when adding one to an order by decoupling the extra logic related to the Coupons Menu. That logic is here added to the `EnhancedCouponListViewController`, while the core functionality is left in the `CouponListViewController`. In the next PR, we will use this basic `CouponListViewController` to show a list of coupons in the context of Order Creation.

We make an exception with the feedback banner, left in the CouponListViewController even if it could be also decoupled. The reason behind this is that it's quite coupled to the list now, which makes removing it from there a considerable effort. Because of this, we found a good compromise by adding a flag  `showFeedbackBannerIfAppropriate` to configure its presentation.

## Testing instructions


1. Go to Menu
2. Open Coupons
3. Test that every functionality related to coupons works as expected: search, adding coupons, removing coupons, empty state.

## Screenshots

N/A

---
